Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Fleet/Driver.hs

```diff
@@ -175,16 +175,20 @@ postDriverFleetAddVehicle ::
   Text ->
   Maybe Text ->
   Maybe Text ->
+  Maybe Common.Role ->
   Common.AddVehicleReq ->
   Flow APISuccess
-postDriverFleetAddVehicle merchantShortId opCity reqDriverPhoneNo requestorId mbFleetOwnerId mbMobileCountryCode req = do
+postDriverFleetAddVehicle merchantShortId opCity reqDriverPhoneNo requestorId mbFleetOwnerId mbMobileCountryCode mbRole req = do
   runRequestValidation Common.validateAddVehicleReq req
   merchant <- findMerchantByShortId merchantShortId
   whenJust mbFleetOwnerId $ \fleetOwnerId -> DCommon.checkFleetOwnerVerification fleetOwnerId merchant.fleetOwnerEnabledCheck
   merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just opCity)
   phoneNumberHash <- getDbHash reqDriverPhoneNo
   let mobileCountryCode = fromMaybe DCommon.mobileIndianCode mbMobileCountryCode
-  entityDetails <- QPerson.findByMobileNumberAndMerchantAndRoles mobileCountryCode phoneNumberHash merchant.id [DP.DRIVER, DP.FLEET_OWNER] >>= fromMaybeM (DriverNotFound reqDriverPhoneNo)
+  let role = case mbRole of
+        Just Common.FLEET -> DP.FLEET_OWNER
+        _ -> DP.DRIVER
```

Comment:
it would be more clear if we throw error in case of operator, or remove it from data declaration
`data Role = DRIVER | FLEET | OPERATOR`